Attack Command Fix| Guard feature working

ZainMGit opened the pull request:
Fixed the attack command problem referenced in this issue.
Issue #6459

- Right clicking a friendly unit while in the attack command now exits instead of attacking
- Previously  guard feature was disabled which is why merge request was denied, now works as intended

ZainMGit commented:
Right clicking is currently the "cancel" button for the attack command, but currently it glitches when hovering a friendly. This change fixes this problem.

efrec commented on luaui/Widgets/cmd_attack_no_ally.lua:
```diff
@@ -1,51 +1,98 @@
+	Spring.GiveOrder(CMD.STOP, {}, {})
+	-- clears the active command cursor 
+	Spring.SetActiveCommand(0)
+end
```
Why not only remove the active command so the guiHandler returns to accepting contextual (rather than active) commands? Why clear the selected unit's command queue?

In other words, can't the entire function be replaced with: `Spring.SetActiveCommand()` or `Spring.SetActiveCommand(nil)`?

ZainMGit commented:
I believe I have addressed all of the comments with this new version.

The local function has been removed and the command queue no longer clears after right clicking. Instead ``Spring.SetActiveCommand(nil)`` is now used. 

The guard portion of the per command logic was not needed but the other logic is to make sure that attack commands on allies aims at ground which ally is standing on. Without this units follow the allied unit around.

efrec commented:
Okay. This is getting there. I think the Attack and Manual Fire orders probably should be treated the same, moving their target to the ground, if possible. There's probably no need for a separate manualfirenoally flag -- but you can search attacknoally in the repo to see where it is being used.

TheChronographer commented:
This needs to be reverted, it's broken all line attack commands.

ZainMGit commented:
> This needs to be reverted, it's broken all line attack commands. 
> 
> 

What do you mean by line attack command? Is this the same as area attack? 
can you send a clip showing this behavior, I will fix this tonight

TheChronographer commented:
> What do you mean by line attack command? Is this the same as area attack? can you send a clip showing this behavior, I will fix this tonight

A(ttack) + right click drag a line of attack orders with multiple units selected. E.g. thugs but it's very commonly used for bombers. With `attack no ally` widget enabled the line attack gets cancelled. disable `attack no ally` and the line attack order goes through. 

Enabled: 
<img width="1050" height="544" alt="image" src="https://github.com/user-attachments/assets/1836384c-5ac0-4f5d-8243-6f1a203a698e" />

Disabled: 
<img width="1626" height="546" alt="image" src="https://github.com/user-attachments/assets/3cc0e198-798c-4339-86f2-8064a4031a4a" />

ZainMGit commented:
Currently working on a fix, problem is related to another widget, custom formations
